pkg/xt/xt.go

```diff
@@ -44,6 +45,12 @@ func Extract(job *Job) {
 
 			start := time.Now()
 
+			// If preserving the file hierarchy, set the output directory to the
+			// folder of the archive being extracted.
+			if job.Preserve {
```

Comment:
_re-writing my comment from last week:_

We should add some code here that does something like:
- remove input path prefix from fileName.
- append fileName Dir to job.Output.
- extract file into job.Output/fileFolder

To keep your current implementation, one would pass the same path to `-o` that was provided to search.

Do you want to tackle this? if not, I'll merge your contribution and do it myself.

Reply:
fwiw, this is how: 
```
file.OutputDir = filepath.Join(job.Output, filepath.Dir(strings.TrimPrefix(folder, fileName)))
```

and change line 30 to 
```
for folder, files := range archives {
```

I'll have a follow contribution tomorrow to tidy this up.